src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,273 @@
+    PRINT_PARAM_STRING("dimension") + "."
+    "\n\n"
+    "So, a simple example where we want to preprocess string the dataset " +
+    PRINT_DATASET("X") + ", which is having string data in its 3 Column."
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "lowercase", 1, "stopwords",
+        1, "punctuation", 1, "dimension", 3) +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess "
+    "dataset.", "o");
+PARAM_STRING_IN("columnDelimiter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d", "\t");
+PARAM_STRING_IN("delimiter", "A set of chars that is used as delimeter to"
+    "tokenize the string dataset ", "D", " ");
+PARAM_STRING_IN("stopwordsfile", "File containing stopwords", "S", "");
+
+PARAM_FLAG("lowercase", "convert to lowercase.", "l");
+PARAM_FLAG("punctuation", "Remove punctuation.", "p");
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_VECTOR_IN_REQ(std::string, "dimension", "Column which contains the"
+    "string data. (1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static vector<vector<string>> CreateDataset(const string& filename,
+                                            const char& columnDelimiter)
+{
+  vector<vector<string>> dataset;
+  // Extracting the Contents of file
+  // File stream.
+  ifstream fin(filename);
+  if (!fin.is_open())
+    Log::Fatal << "Unable to open input file \n";
+  string line, word;
+  while (getline(fin, line))
+  {
+    stringstream streamLine(line);
+    dataset.emplace_back();
+    // delimeter[0] becase the standard function accepts char as input.
+    while (getline(streamLine, word, columnDelimiter))
+    {
+      dataset.back().push_back(word);
+    }
+  }
+  return dataset;
+}
+
+static bool IsNumber(const string& column)
+{
+  for (auto& i : column)
+    if (!isdigit(i))
+      return false;
+  return true;
+}
+
+static unordered_set<size_t> GetColumnIndices(const vector<string>& tempDimension)
+{
+  unordered_set<size_t> dimensions;
+  int columnstartindex, columnendindex;
+  size_t found;
+  for (size_t i = 0; i < tempDimension.size(); i++)
+  {
+    found = tempDimension[i].find('-');
+    if ( found != string::npos)
+    {
+      try
+      {
+        // Has a range include, something of type a-b.
+        string subStringStart = tempDimension[i].substr(0, found);
+        string subStringEnd = tempDimension[i].substr(found+1,
+            tempDimension[i].length());
+        IsNumber(subStringStart);
+        IsNumber(subStringEnd);        
```

Comment:
You forgot to check the return value. You should call `Log::Fatal` if `IsNumber()` returns `false`.